classes/ActionScheduler_FinishedAction.php

```diff
@@ -3,16 +3,19 @@
 /**
  * Class ActionScheduler_FinishedAction
  *
- * Stored action which has finished, which could be because it was cancelled, completed or failed.
+ * Deprecated to avoid ambiguity with stored actions and objects instantiated purely to call the hook.
+ *
+ * @deprecated 1.6.0
  */
-class ActionScheduler_FinishedAction extends ActionScheduler_StoredAction {
+class ActionScheduler_FinishedAction extends ActionScheduler_Action {
 
-	public function execute() {
-		// don't execute
+	public function __construct( $hook, array $args = array(), ActionScheduler_Schedule $schedule = NULL, $group = '' ) {
+		trigger_error( sprintf( __('%1$s is <strong>deprecated</strong> since version 1.6.0! Use an instance of ActionScheduler_NonexecutableAction instead.'), __CLASS__ ) );
+		parent::__construct( $hook, $args, $schedule, $group );
+		$this->is_finished = true;
 	}
 
-	public function is_finished() {
-		return TRUE;
+	public function execute() {
+		// don't execute
```

Comment:
With an additional if statement I mentioned in the `execute()` method above, we wouldn't need to override the method here.

Reply:
That's true, and possibly a good justification for going with a new `run()` method and/or deprecating `execute()` as discussed in the other comment about maintaining backward compatibility.